src/libs/Navigation/AppNavigator/createCustomPlatformStackFullScreenNavigator/CustomFullScreenRouter.tsx

```diff
@@ -0,0 +1,77 @@
+import type {ParamListBase, PartialState, Router, RouterConfigOptions} from '@react-navigation/native';
+import {StackRouter} from '@react-navigation/native';
+import getIsNarrowLayout from '@libs/getIsNarrowLayout';
+import type {PlatformStackNavigationState, PlatformStackRouterOptions} from '@libs/Navigation/PlatformStackNavigation/types';
+import SCREENS from '@src/SCREENS';
+
+type StackState = PlatformStackNavigationState<ParamListBase> | PartialState<PlatformStackNavigationState<ParamListBase>>;
+
+const isAtLeastOneInState = (state: StackState, screenName: string): boolean => state.routes.some((route) => route.name === screenName);
+
+function adaptStateIfNecessary(state: StackState) {
+    const isNarrowLayout = getIsNarrowLayout();
+    const workspaceCentralPane = state.routes.at(-1);
+
+    // There should always be WORKSPACE.INITIAL screen in the state to make sure go back works properly if we deeplinkg to a subpage of settings.
+    if (!isAtLeastOneInState(state, SCREENS.WORKSPACE.INITIAL)) {
+        // @ts-expect-error Updating read only property
+        // noinspection JSConstantReassignment
+        state.stale = true; // eslint-disable-line
```

Comment:
@fabioh8010 this is a copy.paste code as well 🙈 

Honestly I wouldn't make any modifications as of now, since we simply move the code from one place to another 🙃

Reply:
Answered [here](https://github.com/Expensify/App/pull/37891#discussion_r1793699684)